apps/console/src/components/pages/protected/policies/bulk-edit/bulk-edit-policies.tsx

```diff
@@ -0,0 +1,244 @@
+
+  const { fields, append, update, replace, remove } = useFieldArray({
+    control,
+    name: 'fieldsArray',
+    rules: { maxLength: 4 },
+  })
+
+  const onSubmit = async () => {
+    const ids = selectedPolicies.map((policy) => policy.id)
+    const input: Record<string, string | boolean> = {}
+    if (watchedFields.length === 0) return
+
+    if (ids.length === 0) return
+    watchedFields.forEach((field) => {
+      const key = field.selectedObject?.name
+      if (!key) return
+
+      if (field?.selectedValue && field?.value) {
+        input[key] = field.selectedValue
+      }
+
+      if (field.selectedObject?.name === 'policyType' && !field?.selectedValue) {
+        input['clearPolicyType'] = true
+      }
+    })
+
+    try {
+      await bulkEditPolicies({
+        ids: ids,
+        input,
+      })
+      successNotification({
+        title: 'Successfully bulk updated selected policies.',
+      })
+      setIsBulkEditing(false)
+      setSelectedPolicies([])
+      setOpen(false)
+    } catch (error: unknown) {
+      let errorMessage: string | undefined
+      if (error instanceof ClientError) {
+        errorMessage = parseErrorMessage(error.response.errors)
+      }
+      errorNotification({
+        title: errorMessage ?? 'Failed to bulk edit policy. Please try again.',
+      })
+    }
+  }
+
+  return (
+    <Dialog open={open} onOpenChange={setOpen}>
+      <FormProvider {...form}>
+        <DialogTrigger asChild>
+          <Button disabled={selectedPolicies.length === 0} icon={<Pencil />} iconPosition="left" variant="outline">
+            {selectedPolicies && selectedPolicies.length > 0 ? `Bulk Edit (${selectedPolicies.length})` : 'Bulk Edit'}
+          </Button>
+        </DialogTrigger>
+        <form onSubmit={handleSubmit(onSubmit)}>
+          <DialogContent onOpenAutoFocus={(e) => e.preventDefault()} className="max-w-[580px]">
+            <DialogHeader>
+              <DialogTitle>Bulk edit</DialogTitle>
+            </DialogHeader>
+            <div className="flex flex-col gap-4 mt-4">
+              {fields.map((item, index) => {
+                return (
+                  <div key={item.id} className="flex justify-items items-start gap-2">
+                    <div className="flex flex-col items-start gap-2">
+                      <Select
+                        value={watchedFields[index].value || undefined}
+                        onValueChange={(value) => {
+                          const selectedEnum = value as SelectOption
+                          update(index, { value: selectedEnum, selectedObject: allOptionSelects.find((item) => item.selectOptionEnum === selectedEnum), selectedValue: undefined })
+                        }}
+                      >
+                        <SelectTrigger className="w-48">
+                          <SelectValue placeholder="Select a field to update" />
```

Comment:
(Make sure the control placeholder matches too)
```suggestion
                          <SelectValue placeholder="Select field..." />
```